Move wallet tables to a dedicated DB

IvanBelyakoff opened the pull request:
## What has been done:

- created a dedicated DB for wallet next to app db with scheme `<keyUID>-wallet.db` and the same password as app DB
- moved wallet-related tables to a dedicated db:
  - blocks
  - blocks_ranges
  - blocks_ranges_sequential
  - pending_transactions
  - saved_addresses
  - token_balances
  - tokens
  - visible_tokens
  - currency_format_cache
  - multi_transactions
  - balance_history
  - price_cache
  - collectibles_ownership_cache
  - transfers

- test fixes:
  - moved test db setup to a common place 't/helpers'
  - created interface for initializing db, which is implemented for appdatabase and
walletdatabase. TBD for multiaccounts DB.
  - unified DB initialisation for all tests using helpers and new interface.
  - reduced sqlcipher kdf iterations for all tests to `1` to improve performance

- sqlite migration improvements. While working on the feature, I tried to do in-code migration first, which led to some refactoring of migration functions. I dropped in-code migration in favor of sql files, because it did not allow standalone wallet db instantiation for tests, but I kept some related code for improvement purposes.
  - some functions split for more cohesion
  - custom PostSteps are stored as pointers to allow their retrieval and parameters passing
in runtime if needed (some extra work is dropped and TBD when needed)

### NOTE:
Data is not copied from app db to wallet db. This mostly is due to some bug (probably in sqlcipher/go-sqlcipher) that caused an error on detaching an attached database after copying data.

Please review single commit at a time for clarity.

Closes [#10964](https://github.com/status-im/status-desktop/issues/10964)

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (150)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 79a49559 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/1/) | 2023-07-24 10:58:05 | ~1 min | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/1/consoleText) |
| :x: | 79a49559 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/1/) | 2023-07-24 10:58:10 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/1/consoleText) |
| :x: | 79a49559 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/1/) | 2023-07-24 10:58:14 | ~1 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/1/consoleText) |
| :heavy_multiplication_x: | 79a49559 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/1/) | 2023-07-24 10:59:24 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/1//consoleText) |
| | | | | | | |
| :x: | a2a0e58f | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/2/) | 2023-07-24 11:07:06 | ~13 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/2/consoleText) |
| :x: | a2a0e58f | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/2/) | 2023-07-24 11:07:11 | ~17 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/2/consoleText) |
| :heavy_multiplication_x: | a2a0e58f | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/2/) | 2023-07-24 11:07:59 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/2//consoleText) |
| :x: | a2a0e58f | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/2/) | 2023-07-24 11:08:06 | ~1 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/2/consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | b2b8307e | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/3/) | 2023-08-02 13:37:28 | ~50 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/3//consoleText) |
| :heavy_check_mark: | b2b8307e | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/3/) | 2023-08-02 13:39:50 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230802-133637-b2b830-pr3793.zip) |
| :heavy_check_mark: | b2b8307e | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/3/) | 2023-08-02 13:40:42 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230802-133636-b2b830-pr3793.zip) |
| :heavy_check_mark: | b2b8307e | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/3/) | 2023-08-02 13:41:24 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230802-133635-b2b830-pr3793.aar) |
| | | | | | | |
| :heavy_multiplication_x: | d362bacb | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/4/) | 2023-08-02 14:33:18 | ~38 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/4//consoleText) |
| :heavy_check_mark: | d362bacb | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/4/) | 2023-08-02 14:33:56 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230802-143239-d362ba-pr3793.zip) |
| :heavy_check_mark: | d362bacb | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/4/) | 2023-08-02 14:35:48 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230802-143239-d362ba-pr3793.zip) |
| :heavy_multiplication_x: | d362bacb | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/5/) | 2023-08-02 14:35:57 | ~35 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/5//consoleText) |
| :heavy_check_mark: | d362bacb | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/4/) | 2023-08-02 14:37:56 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230802-143239-d362ba-pr3793.aar) |
| | | | | | | |
| :heavy_check_mark: | cd9a1999 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/5/) | 2023-08-02 14:44:34 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230802-144136-cd9a19-pr3793.zip) |
| :heavy_check_mark: | cd9a1999 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/5/) | 2023-08-02 14:45:47 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230802-144134-cd9a19-pr3793.aar) |
| :heavy_multiplication_x: | cd9a1999 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/6/) | 2023-08-02 14:46:45 | ~5 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/6//consoleText) |
| :heavy_check_mark: | cd9a1999 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/5/) | 2023-08-02 14:54:08 | ~12 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230802-144135-cd9a19-pr3793.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 88c91fc9 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/7/) | 2023-08-03 13:19:23 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/7//consoleText) |
| :heavy_check_mark: | 88c91fc9 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/6/) | 2023-08-03 13:19:38 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230803-131638-88c91f-pr3793.zip) |
| :heavy_check_mark: | 88c91fc9 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/6/) | 2023-08-03 13:21:16 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230803-131635-88c91f-pr3793.aar) |
| :heavy_check_mark: | 88c91fc9 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/6/) | 2023-08-03 13:21:36 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230803-131638-88c91f-pr3793.zip) |
| | | | | | | |
| :heavy_check_mark: | 0d759a16 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/7/) | 2023-08-03 14:14:26 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230803-141248-0d759a-pr3793.zip) |
| :heavy_check_mark: | 0d759a16 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/7/) | 2023-08-03 14:14:38 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230803-141244-0d759a-pr3793.aar) |
| :heavy_multiplication_x: | 0d759a16 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/8/) | 2023-08-03 14:15:04 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/8//consoleText) |
| :heavy_check_mark: | 0d759a16 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/7/) | 2023-08-03 14:18:53 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230803-141248-0d759a-pr3793.zip) |
| | | | | | | |
| :heavy_multiplication_x: | b10e853c | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/9/) | 2023-08-03 15:13:56 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/9//consoleText) |
| :heavy_check_mark: | b10e853c | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/8/) | 2023-08-03 15:14:32 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230803-151244-b10e85-pr3793.aar) |
| :heavy_check_mark: | b10e853c | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/8/) | 2023-08-03 15:16:01 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230803-151247-b10e85-pr3793.zip) |
| :heavy_check_mark: | b10e853c | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/8/) | 2023-08-03 15:16:33 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230803-151247-b10e85-pr3793.zip) |
| | | | | | | |
| :x: | e1d05ae9 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/9/) | 2023-08-03 15:21:24 | ~14 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/9/consoleText) |
| :x: | e1d05ae9 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/9/) | 2023-08-03 15:21:24 | ~19 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/9/consoleText) |
| :x: | e1d05ae9 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/9/) | 2023-08-03 15:21:26 | ~21 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/9/consoleText) |
| :heavy_multiplication_x: | e1d05ae9 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/10/) | 2023-08-03 15:23:09 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/10//consoleText) |
| | | | | | | |
| :heavy_check_mark: | c80c98a5 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/10/) | 2023-08-03 16:06:37 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230803-160504-c80c98-pr3793.zip) |
| :heavy_check_mark: | c80c98a5 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/10/) | 2023-08-03 16:06:51 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230803-160500-c80c98-pr3793.aar) |
| :heavy_multiplication_x: | c80c98a5 | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/11/) | 2023-08-03 16:07:32 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/11//consoleText) |
| :heavy_check_mark: | c80c98a5 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/10/) | 2023-08-03 16:08:19 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230803-160504-c80c98-pr3793.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 7b6c4c3e | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/12/) | 2023-08-03 16:14:55 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/12//consoleText) |
| :heavy_check_mark: | 7b6c4c3e | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/11/) | 2023-08-03 16:17:22 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230803-161345-7b6c4c-pr3793.zip) |
| :heavy_check_mark: | 7b6c4c3e | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/11/) | 2023-08-03 16:17:36 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230803-161345-7b6c4c-pr3793.zip) |
| :heavy_check_mark: | 7b6c4c3e | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/11/) | 2023-08-03 16:18:49 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230803-161345-7b6c4c-pr3793.aar) |
| | | | | | | |
| :heavy_multiplication_x: | b42a4042 | [#13](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/13/) | 2023-08-03 16:26:38 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/13//consoleText) |
| :heavy_check_mark: | b42a4042 | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/12/) | 2023-08-03 16:26:39 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230803-162529-b42a40-pr3793.zip) |
| :heavy_check_mark: | b42a4042 | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/12/) | 2023-08-03 16:27:20 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230803-162525-b42a40-pr3793.aar) |
| :heavy_check_mark: | b42a4042 | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/12/) | 2023-08-03 16:28:03 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230803-162525-b42a40-pr3793.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 13c4cafc | [#14](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/14/) | 2023-08-03 16:34:08 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/14//consoleText) |
| :heavy_check_mark: | 13c4cafc | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/13/) | 2023-08-03 16:34:32 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230803-163259-13c4ca-pr3793.zip) |
| :heavy_check_mark: | 13c4cafc | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/13/) | 2023-08-03 16:34:33 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230803-163255-13c4ca-pr3793.aar) |
| :heavy_check_mark: | 13c4cafc | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/13/) | 2023-08-03 16:36:05 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230803-163259-13c4ca-pr3793.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 836c60a1 | [#15](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/15/) | 2023-08-03 16:37:08 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/15//consoleText) |
| :heavy_check_mark: | 836c60a1 | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/14/) | 2023-08-03 16:37:22 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230803-163555-836c60-pr3793.aar) |
| :heavy_check_mark: | 836c60a1 | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/14/) | 2023-08-03 16:37:24 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230803-163555-836c60-pr3793.zip) |
| :heavy_check_mark: | 836c60a1 | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/14/) | 2023-08-03 16:39:04 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230803-163610-836c60-pr3793.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 74f8e96e | [#16](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/16/) | 2023-08-03 16:39:36 | ~52 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/16//consoleText) |
| :heavy_check_mark: | 74f8e96e | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/15/) | 2023-08-03 16:40:00 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230803-163840-74f8e9-pr3793.zip) |
| :heavy_check_mark: | 74f8e96e | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/15/) | 2023-08-03 16:40:04 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230803-163840-74f8e9-pr3793.aar) |
| :heavy_check_mark: | 74f8e96e | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/15/) | 2023-08-03 16:42:09 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230803-163907-74f8e9-pr3793.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 8acd031c | [#17](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/17/) | 2023-08-03 16:53:30 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/17//consoleText) |
| :heavy_check_mark: | 8acd031c | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/16/) | 2023-08-03 16:53:36 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230803-165220-8acd03-pr3793.zip) |
| :heavy_check_mark: | 8acd031c | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/16/) | 2023-08-03 16:53:57 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230803-165220-8acd03-pr3793.aar) |
| :heavy_check_mark: | 8acd031c | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/16/) | 2023-08-03 16:54:56 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230803-165220-8acd03-pr3793.zip) |
| | | | | | | |
| :heavy_multiplication_x: | c64e7d33 | [#18](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/18/) | 2023-08-03 17:13:01 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/18//consoleText) |
| :heavy_check_mark: | c64e7d33 | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/17/) | 2023-08-03 17:13:31 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230803-171200-c64e7d-pr3793.aar) |
| :heavy_check_mark: | c64e7d33 | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/17/) | 2023-08-03 17:13:43 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230803-171200-c64e7d-pr3793.zip) |
| :heavy_check_mark: | c64e7d33 | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/17/) | 2023-08-03 17:15:00 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230803-171200-c64e7d-pr3793.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 98165b34 | [#19](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/19/) | 2023-08-03 17:34:21 | ~50 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/19//consoleText) |
| :heavy_check_mark: | 98165b34 | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/18/) | 2023-08-03 17:34:50 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230803-173330-98165b-pr3793.zip) |
| :heavy_check_mark: | 98165b34 | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/18/) | 2023-08-03 17:35:00 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230803-173326-98165b-pr3793.aar) |
| :heavy_check_mark: | 98165b34 | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/18/) | 2023-08-03 17:36:32 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230803-173327-98165b-pr3793.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 797d4f4d | [#20](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/20/) | 2023-08-03 17:40:56 | ~50 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/20//consoleText) |
| :heavy_check_mark: | 797d4f4d | [#19](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/19/) | 2023-08-03 17:41:15 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230803-174005-797d4f-pr3793.zip) |
| :heavy_check_mark: | 797d4f4d | [#19](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/19/) | 2023-08-03 17:41:32 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230803-174000-797d4f-pr3793.aar) |
| :heavy_check_mark: | 797d4f4d | [#19](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/19/) | 2023-08-03 17:42:50 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230803-174005-797d4f-pr3793.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 8a6e3560 | [#21](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/21/) | 2023-08-03 17:43:23 | ~48 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/21//consoleText) |
| :heavy_check_mark: | 8a6e3560 | [#20](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/20/) | 2023-08-03 17:43:48 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230803-174234-8a6e35-pr3793.zip) |
| :heavy_check_mark: | 8a6e3560 | [#20](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/20/) | 2023-08-03 17:43:58 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230803-174230-8a6e35-pr3793.aar) |
| :heavy_check_mark: | 8a6e3560 | [#20](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/20/) | 2023-08-03 17:45:29 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230803-174254-8a6e35-pr3793.zip) |
| | | | | | | |
| :heavy_check_mark: | f13abe76 | [#21](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/21/) | 2023-08-03 17:53:33 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230803-175220-f13abe-pr3793.zip) |
| :heavy_check_mark: | f13abe76 | [#21](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/21/) | 2023-08-03 17:53:57 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230803-175220-f13abe-pr3793.aar) |
| :heavy_multiplication_x: | f13abe76 | [#22](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/22/) | 2023-08-03 17:54:40 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/22//consoleText) |
| :heavy_check_mark: | f13abe76 | [#21](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/21/) | 2023-08-03 17:54:57 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230803-175220-f13abe-pr3793.zip) |
| | | | | | | |
| :heavy_check_mark: | d559d0ae | [#22](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/22/) | 2023-08-04 11:40:45 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230804-113924-d559d0-pr3793.zip) |
| :heavy_check_mark: | d559d0ae | [#22](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/22/) | 2023-08-04 11:41:04 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230804-113920-d559d0-pr3793.aar) |
| :heavy_check_mark: | d559d0ae | [#22](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/22/) | 2023-08-04 11:41:57 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230804-113920-d559d0-pr3793.zip) |
| :heavy_multiplication_x: | d559d0ae | [#23](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/23/) | 2023-08-04 12:00:35 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/23//consoleText) |
| | | | | | | |
| :heavy_check_mark: | f67c6a1b | [#23](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/23/) | 2023-08-04 14:57:20 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230804-145552-f67c6a-pr3793.zip) |
| :heavy_check_mark: | f67c6a1b | [#23](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/23/) | 2023-08-04 14:57:41 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230804-145552-f67c6a-pr3793.aar) |
| :heavy_check_mark: | f67c6a1b | [#23](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/23/) | 2023-08-04 14:58:34 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230804-145552-f67c6a-pr3793.zip) |
| :heavy_multiplication_x: | f67c6a1b | [#24](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/24/) | 2023-08-04 15:17:10 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/24//consoleText) |
| | | | | | | |
| :heavy_check_mark: | abe5c2ec | [#24](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/24/) | 2023-08-04 15:58:04 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230804-155652-abe5c2-pr3793.zip) |
| :heavy_check_mark: | abe5c2ec | [#24](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/24/) | 2023-08-04 15:58:46 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230804-155652-abe5c2-pr3793.aar) |
| :heavy_check_mark: | abe5c2ec | [#24](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/24/) | 2023-08-04 15:59:41 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230804-155652-abe5c2-pr3793.zip) |
| :heavy_multiplication_x: | abe5c2ec | [#25](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/25/) | 2023-08-04 16:17:07 | ~20 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/25//consoleText) |
| :heavy_check_mark: | abe5c2ec | [#26](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/26/) | 2023-08-04 16:32:51 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/26//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 48601078 | [#27](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/27/) | 2023-08-08 09:48:56 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/27//consoleText) |
| :heavy_check_mark: | 48601078 | [#25](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/25/) | 2023-08-08 09:49:11 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230808-094743-486010-pr3793.zip) |
| :heavy_check_mark: | 48601078 | [#25](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/25/) | 2023-08-08 09:49:23 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230808-094738-486010-pr3793.aar) |
| :heavy_check_mark: | 48601078 | [#25](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/25/) | 2023-08-08 09:52:19 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230808-094743-486010-pr3793.zip) |
| | | | | | | |
| :heavy_check_mark: | 83bf24d3 | [#26](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/26/) | 2023-08-08 09:59:04 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230808-095752-83bf24-pr3793.zip) |
| :heavy_check_mark: | 83bf24d3 | [#26](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/26/) | 2023-08-08 09:59:26 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230808-095748-83bf24-pr3793.aar) |
| :heavy_check_mark: | 83bf24d3 | [#26](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/26/) | 2023-08-08 10:00:16 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230808-095752-83bf24-pr3793.zip) |
| :heavy_multiplication_x: | 83bf24d3 | [#28](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/28/) | 2023-08-08 10:18:40 | ~20 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/28//consoleText) |
| | | | | | | |
| :heavy_check_mark: | dc2688f4 | [#27](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/27/) | 2023-08-08 10:25:32 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230808-102412-dc2688-pr3793.zip) |
| :heavy_multiplication_x: | dc2688f4 | [#29](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/29/) | 2023-08-08 10:25:49 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/29//consoleText) |
| :heavy_check_mark: | dc2688f4 | [#27](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/27/) | 2023-08-08 10:25:57 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230808-102408-dc2688-pr3793.aar) |
| :heavy_check_mark: | dc2688f4 | [#27](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/27/) | 2023-08-08 10:26:56 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230808-102412-dc2688-pr3793.zip) |
| | | | | | | |
| :heavy_check_mark: | 54a19f8f | [#28](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/28/) | 2023-08-08 10:36:17 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230808-103513-54a19f-pr3793.zip) |
| :heavy_check_mark: | 54a19f8f | [#28](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/28/) | 2023-08-08 10:36:28 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230808-103509-54a19f-pr3793.aar) |
| :heavy_check_mark: | 54a19f8f | [#28](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/28/) | 2023-08-08 10:37:27 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230808-103509-54a19f-pr3793.zip) |
| :heavy_check_mark: | 54a19f8f | [#30](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/30/) | 2023-08-08 10:39:54 | ~4 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/30//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 18075e83 | [#29](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/29/) | 2023-08-08 11:23:55 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230808-112239-18075e-pr3793.zip) |
| :heavy_multiplication_x: | 18075e83 | [#31](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/31/) | 2023-08-08 11:24:14 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/31//consoleText) |
| :heavy_check_mark: | 18075e83 | [#29](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/29/) | 2023-08-08 11:24:22 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230808-112239-18075e-pr3793.aar) |
| :heavy_check_mark: | 18075e83 | [#29](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/29/) | 2023-08-08 11:25:12 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230808-112239-18075e-pr3793.zip) |
| | | | | | | |
| :heavy_multiplication_x: | f55fec5e | [#32](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/32/) | 2023-08-08 11:58:35 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/32//consoleText) |
| :heavy_check_mark: | f55fec5e | [#30](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/30/) | 2023-08-08 12:00:17 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230808-115613-f55fec-pr3793.zip) |
| :heavy_check_mark: | f55fec5e | [#30](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/30/) | 2023-08-08 12:01:39 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230808-115610-f55fec-pr3793.aar) |
| :heavy_check_mark: | f55fec5e | [#30](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/30/) | 2023-08-08 12:04:25 | ~8 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230808-115610-f55fec-pr3793.zip) |
| :heavy_multiplication_x: | f55fec5e | [#33](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/33/) | 2023-08-08 12:10:01 | ~5 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/33//consoleText) |
| :heavy_check_mark: | f55fec5e | [#34](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/34/) | 2023-08-08 12:33:51 | ~18 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/34//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 5be3d117 | [#31](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/31/) | 2023-08-09 09:45:34 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230809-094414-5be3d1-pr3793.zip) |
| :heavy_check_mark: | 5be3d117 | [#31](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/31/) | 2023-08-09 09:46:02 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230809-094414-5be3d1-pr3793.aar) |
| :heavy_check_mark: | 5be3d117 | [#31](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/31/) | 2023-08-09 09:46:42 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230809-094414-5be3d1-pr3793.zip) |
| :heavy_check_mark: | 5be3d117 | [#35](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/35/) | 2023-08-09 10:05:51 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/35//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | bd48d449 | [#36](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/36/) | 2023-08-10 16:12:05 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/36//consoleText) |
| :heavy_check_mark: | bd48d449 | [#32](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/32/) | 2023-08-10 16:12:09 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230810-161053-bd48d4-pr3793.zip) |
| :heavy_check_mark: | bd48d449 | [#32](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/32/) | 2023-08-10 16:12:35 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230810-161053-bd48d4-pr3793.aar) |
| :heavy_check_mark: | bd48d449 | [#32](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/32/) | 2023-08-10 16:13:24 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230810-161053-bd48d4-pr3793.zip) |
| | | | | | | |
| :heavy_check_mark: | 90a49811 | [#33](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/33/) | 2023-08-10 16:22:29 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230810-162127-90a498-pr3793.zip) |
| :heavy_check_mark: | 90a49811 | [#33](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/33/) | 2023-08-10 16:22:42 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230810-162123-90a498-pr3793.aar) |
| :heavy_check_mark: | 90a49811 | [#33](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/33/) | 2023-08-10 16:23:33 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230810-162123-90a498-pr3793.zip) |
| :heavy_check_mark: | 90a49811 | [#37](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/37/) | 2023-08-10 16:41:53 | ~20 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/37//consoleText) |
| | | | | | | |
| :heavy_check_mark: | f12a1f04 | [#34](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/34/) | 2023-08-10 16:48:32 | ~58 sec | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230810-164733-f12a1f-pr3793.zip) |
| :heavy_check_mark: | f12a1f04 | [#34](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/34/) | 2023-08-10 16:48:43 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230810-164733-f12a1f-pr3793.aar) |
| :heavy_check_mark: | f12a1f04 | [#34](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/34/) | 2023-08-10 16:49:46 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230810-164733-f12a1f-pr3793.zip) |
| :heavy_check_mark: | f12a1f04 | [#38](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/38/) | 2023-08-10 16:50:57 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/38//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 37ce3cb2 | [#35](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/35/) | 2023-08-10 17:04:07 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230810-170253-37ce3c-pr3793.zip) |
| :heavy_check_mark: | 37ce3cb2 | [#35](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/35/) | 2023-08-10 17:04:30 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230810-170253-37ce3c-pr3793.aar) |
| :heavy_check_mark: | 37ce3cb2 | [#35](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/35/) | 2023-08-10 17:05:10 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230810-170253-37ce3c-pr3793.zip) |
| :heavy_check_mark: | 37ce3cb2 | [#39](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/39/) | 2023-08-10 17:21:52 | ~18 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/39//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 5fab5947 | [#36](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/36/) | 2023-08-11 11:53:06 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230811-115007-5fab59-pr3793.zip) |
| :heavy_check_mark: | 5fab5947 | [#36](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/36/) | 2023-08-11 11:53:44 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230811-115007-5fab59-pr3793.zip) |
| :heavy_check_mark: | 5fab5947 | [#36](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/36/) | 2023-08-11 11:54:46 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230811-115003-5fab59-pr3793.aar) |
| :heavy_multiplication_x: | 5fab5947 | [#40](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/40/) | 2023-08-11 12:01:34 | ~11 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/40//consoleText) |
| :heavy_multiplication_x: | 5fab5947 | [#41](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/41/) | 2023-08-11 12:17:13 | ~10 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/41//consoleText) |
| :heavy_check_mark: | 5fab5947 | [#42](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/42/) | 2023-08-11 13:15:19 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/42//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | fe5dba1e | [#37](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/37/) | 2023-08-14 11:40:23 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230814-113858-fe5dba-pr3793.zip) |
| :heavy_check_mark: | fe5dba1e | [#37](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/37/) | 2023-08-14 11:40:53 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230814-113858-fe5dba-pr3793.aar) |
| :heavy_check_mark: | fe5dba1e | [#37](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/37/) | 2023-08-14 11:42:10 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230814-113858-fe5dba-pr3793.zip) |
| :heavy_check_mark: | fe5dba1e | [#43](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/43/) | 2023-08-14 12:06:05 | ~27 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/43//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 700d1659 | [#38](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3793/38/) | 2023-08-17 16:09:36 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230817-160704-700d16-pr3793.aar) |
| :heavy_check_mark: | 700d1659 | [#38](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3793/38/) | 2023-08-17 16:09:42 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230817-160704-700d16-pr3793.zip) |
| :heavy_check_mark: | 700d1659 | [#38](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3793/38/) | 2023-08-17 16:11:01 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230817-160704-700d16-pr3793.zip) |
| :heavy_check_mark: | 700d1659 | [#47](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/47/) | 2023-08-17 17:42:27 | ~22 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3793/47//consoleText) |

IvanBelyakoff commented on appdatabase/database.go:
```diff
@@ -487,3 +507,211 @@ func migrateWalletTransferFromToAddresses(sqlTx *sql.Tx) error {
+		query := fmt.Sprintf("INSERT INTO %s.%s SELECT * FROM main.%s;", walletDbName, tableName, tableName)
+
+		fmt.Println("query start: ", query)
+		_, err := sqlTx.Exec(query)
```
This function fails. I thought my test DB (it is my real status app DB) might have too many records, but I tried to use `LIMIT 10` and `OFFSET`, but it did not help - same error `out of memory` during the first iteration.

```
createDatabaseSchema, query:
createDatabaseSchema end
AttachDatabaseWithDefaultSettings start
database is attached success
copyDataToNewTables start
query start:  INSERT INTO wallet.blocks SELECT * FROM main.blocks;
query failed:  INSERT INTO wallet.blocks SELECT * FROM main.blocks;  err: out of memory
    /home/blade/work/projects/Status/status-desktop/vendor/status-go/appdatabase/database_test.go:520:
        	Error Trace:	/home/blade/work/projects/Status/status-desktop/vendor/status-go/appdatabase/database_test.go:520
        	Error:      	Received unexpected error:
        	            	failed to rollback migration to version 1689248269: transaction start failed in line 0:  (details: out of memory)
        	Test:       	Test_migrateWalletTablesToDedicatedDB
--- FAIL: Test_migrateWalletTablesToDedicatedDB (1.71s)
FAIL
FAIL	github.com/status-im/status-go/appdatabase	1.732s
```

stefandunca commented on appdatabase/database.go:
```diff
@@ -487,3 +507,213 @@ func migrateWalletTransferFromToAddresses(sqlTx *sql.Tx) error {
+}
+
+// func createAllTablesQueries(sqlTx *sql.Tx, whereClause string) ([]string, error) {
+func createAllTablesQueries(sqlTx *sql.Tx, tableNames []string) ([]string, error) {
```
@IvanBelyakoff I think this table specs copying automation is great. How will the automated copying of the primary key auto-index work? You can't create a "manual" index until you have the tables. However, when you create a table, SQLite automatically indexes the primary key (composed or not). What happens when we copy the previously automatically created indexes?

IvanBelyakoff replied:
Stefan, my understanding is that "CREATE INDEX" command will be executed to create an index on table and once data will be copied to the wallet db, the database will index the data. Is it not correct in your opinion? I don't think I copy the indices. I copy the commands to create tables and indices. Then I use the same commands for the new DB.

stefandunca replied:
I just checked my DB `sqlite_master`, and I see that the `sql` misses the index entries listed by running `PRAGMA index_list('<table_name>');`. Maybe see if there is a good way to keep this using the introspection method.

IvanBelyakoff replied:
@stefandunca Good point, indeed the indices were missing with the command, that I have in code. I actually noticed that, but did not get to it yet, while fixing other issues. The working command for me:
sqlite> SELECT sql FROM sqlite_master WHERE type='index' AND tbl_name="transfers";
```
CREATE INDEX idx_transfers_blk_loaded ON transfers(blk_number, loaded)
CREATE INDEX idx_transfers_filter ON transfers (status, token_address, token_id)
sqlite> 
sqlite> pragma index_list("transfers");
0|idx_transfers_filter|0|c|0
1|idx_transfers_blk_loaded|0|c|0
2|sqlite_autoindex_transfers_1|1|u|0
```
I believe `index_list` does not list auto indices, not created manually. But I will double check with tests that the indices on source and destination DBs match

stefandunca commented on walletdatabase/migrations/sql_queries/0022_pending_transfers.up.sql:
```diff
@@ -0,0 +1,22 @@
+ALTER TABLE pending_transactions RENAME TO pending_transactions_old;
```
How about having only the initial migration that creates the current wallet tables state and one custom migration to copy the data?

IvanBelyakoff replied:
omg, sorry for confusion, this is a drafty draft only intended to ask you teammates, if you spot any obvious issue why I get out of mem issue, so this was committed by mistake, it wont be there

stefandunca reviewed (commented):
You did a great job of figuring out how to migrate data from one table to another. However, can't see a reason for the out-of-memory error. It might be down to how SQLite manages the memory when mapping two DBs. Or maybe a limitation of some sort with sqlchiper?

Following your work, I think it would be best to manually create the tables using the new wallet migrations scripts in one go. Running the old migrations on the new DB given that we don't have data might be pointless or confusing. This will be some extra work from your side, but as a result, we will have it documented properly in the first new wallet migration scripts.

The scope of custom migration is to do only what is not possible with the migration scripts: copy the data.

In short, I would expect that this custom migration only copies the data and creation of tables and indexes is visible in the first migration script.

IvanBelyakoff commented:
@stefandunca thank you for your feedback. It is great to receive it at this quite early point. 
I considered the default migration way using sql files, but decided to do this migration in code for several reasons:
- automatic extraction of table creation schemas: While working on this PR, I might miss some changes/migrations of tables, and will need to maintain and update them while working and rebase as soon as some new migration appears - that is troublesome - whether I will have to extract and combine all the migration sql files in a single one of copy all the related sql files to a separate `wallet` directory, which in my opinion only pollutes the project and it is still hard to track all the changes, while we need only the latest ones . With in-code approach I receive always the latest schema for all the tables and the only thing I need to keep track of is my own migration files timestamp.
- Instead of running all the migrations sql file to get to the final version, I run only single CREATE command for each table which is faster, more precise and cleaner approach.
- Splitting this operation into several steps - sql migration files and in-code requires more work, harder to understand and imho does not worth the efforts.

Though I have not quite understoodd the drawbacks that you mentioned of this way of doing it, sorry. If you could elaborate more on that, that would be awesome. We could have a call instead.

stefandunca commented:
> * and will need to maintain and update them while working and rebase as soon *

I agree that the small benefits to have the current configuration snapshot might not be worth the effort though.

> * Instead of running all the migrations sql file to get to the final version, I run only single CREATE command for each table which is faster, more precise and cleaner approach.

I'm not very familiar with this approach so great to understand from you that it is consistent and complete

IvanBelyakoff committed:
feat(migration): sqlite migration improvements.
Some functions split to be more cohesive, custom PostSteps are
stored as pointers to allow their retrieval and parameters passing
in runtime if needed (some extra work is dropped and TBD
when needed)

alaibe commented on walletdatabase/migrations/sql/doc.go:
```diff
@@ -0,0 +1,3 @@
+package sql
+
+//go:generate go-bindata -pkg migrations -o ../bindata.go ./
```
shall you modify the makefile too that it generate this?

IvanBelyakoff replied:
done, thank you

dlipicar commented on appdatabase/migrations/sql/1690734500_move_wallet_tables_to_wallet_db.down.sql:
```diff
@@ -0,0 +1,160 @@
+network_id UNSIGNED BIGINT NOT NULL,
+address VARCHAR NOT NULL,
+blk_from BIGINT NOT NULL,
+blk_to BIGINT NOT NULL
```
weird formatting here

IvanBelyakoff replied:
Yeah, I actually had a piece of code that just select all create table statements from app db and dumps them into a file. So the formatting comes from the sql files originally. I did not touch it to avoid reformatting on each rebase. But I can do that before I merge

dlipicar commented on appdatabase/migrations/sql/1690734500_move_wallet_tables_to_wallet_db.down.sql:
```diff
@@ -0,0 +1,160 @@
+  PRIMARY KEY (network_id, hash)
+) WITHOUT ROWID;
+
+CREATE TABLE "saved_addresses" (
```
I'm guessing the quotes do nothing here, I think it's the first time I've seen them used

IvanBelyakoff replied:
That is right, it does nothing, just taken from sql migration file automatically

dlipicar commented on appdatabase/migrations/sql/1690734500_move_wallet_tables_to_wallet_db.down.sql:
```diff
@@ -0,0 +1,160 @@
+    to_asset VARCHAR NOT NULL,
+    type VARCHAR NOT NULL,
+    timestamp UNSIGNED BIGINT NOT NULL
+, to_amount VARCHAR, from_network_id UNSIGNED BIGINT, to_network_id UNSIGNED BIGINT, cross_tx_id VARCHAR DEFAULT "", from_tx_hash BLOB, to_tx_hash BLOB);
```
not sure if this is github misbehaving, seeing weird formatting here
![image](https://github.com/status-im/status-go/assets/11161531/ba566f64-7372-45d1-8095-7ca2b41d1d79)

dlipicar commented on appdatabase/migrations/sql/1690734500_move_wallet_tables_to_wallet_db.down.sql:
```diff
@@ -0,0 +1,160 @@
+);
+
+CREATE TABLE transfers (
+network_id UNSIGNED BIGINT NOT NULL,
```
no indentation, as opposed to all other create statements

dlipicar commented on appdatabase/migrations/sql/1690734500_move_wallet_tables_to_wallet_db.up.sql:
```diff
@@ -0,0 +1,18 @@
+DROP TABLE pending_transactions;
+DROP TABLE saved_addresses;
+-- token_balances is the only table that was placed by mistake in nodeconfig
+-- migrations and in tests it is missing if nodeconfig migration is not used.
```
OOC, what's supposed to go in the nodeconfig migrations?

IvanBelyakoff replied:
Oh, I should fix the comment. `token_balances` migration file was placed in protocol migrations, not in nodeconfig migrations. As I see prevnodeconfig migration files date earlier and are now superseeded by plain migrations folder. But not sure about `protocol` migrations - probably just related to some waku-related things judging by the code, but in the end, the files are merged and applied in correct order. But in tests if you want to achieve the same, you should do some work I believe

dlipicar commented on services/wallet/activity/activity.go:
```diff
@@ -896,3 +907,28 @@ func contractTypeFromDBType(dbType string) (transferType *TransferType) {
+	}
+
+	for _, address := range addresses {
+		_, err = db.Exec(fmt.Sprintf("INSERT OR IGNORE INTO %s (address) VALUES (?)", keypairAccountsTable), address)
```
possibly more efficient to do a Begin..Commit here

dlipicar reviewed (approved):
awesome work!

IvanBelyakoff committed:
tests: Added unit test for copying 'keypairs_accounts' table to the wallet db.
Added a unit test for changing app and wallet DBs passwords.
Refactored geth_backend to simplify and allow wallet db password changing.
Fixed opening database with wrong password.

IvanBelyakoff committed:
feat(wallet): move wallet-related tables to a dedicated db.
The only place where appDB is used in wallet is activity,
which refers to `keycards_accounts` table. So a temporary
table `keycards_accounts` is created in wallet db and updated
before each activity query.

IvanBelyakoff committed:
fix(tests): moved test db setup to a common place 't/helpers', created
interface for initializing db, which is implemented for appdatabase and
walletdatabase. TBD for multiaccounts DB.
Unified DB initializion for all tests using helpers and new interface.
Reduced sqlcipher kdf iterations for all tests to 1.

stefandunca commented on Makefile:
```diff
@@ -392,6 +392,10 @@ migration:
 migration-check:
 	bash _assets/scripts/migration_check.sh
 
+migration-wallet: DEFAULT_WALLET_MIGRATION_PATH := walletdatabase/migrations/sql
```
Wondering why two entries for `migration-wallet `

IvanBelyakoff replied:
@stefandunca I dont understand this comment. Can you elaborate more?

stefandunca replied:
I meant `migration-wallet`, wrong copy-pasta

IvanBelyakoff replied:
Stefan, I am not an expert on make, but I remember trying to combine setting a variable and the next `touch` command into a single line or under a single target, but it did not work for me. So probably that is how it should be done to make it work. But if you know how to change that, please share

stefandunca replied:
I have no clue of make, just wondering :) Thanks for clarifying.

stefandunca commented on services/wallet/activity/activity.go:
```diff
@@ -655,7 +659,14 @@ func getActivityEntries(ctx context.Context, deps FilterDependencies, addresses
-	queryString := fmt.Sprintf(queryFormatString, involvedAddresses, toAddresses, assetsTokenCodes, assetsERC20, networks,
+	// Since the filter query needs addresses which are in a different database, we need to update the
+	// keypairs_accounts table in the current database with the latest addresses from the accounts database
+	err := updateKeypairsAccountsTable(deps.accountsDb, deps.db)
```
A good workaround for now, however, will be nice to have a task, for follow-up, linked to this change and remove this requirement entirely and rely on the filtered addresses as we discussed in private.

IvanBelyakoff replied:
https://github.com/status-im/status-go/issues/3881
Thanx for the reminder, Stefan!

stefandunca commented on sqlite/migrate.go:
```diff
@@ -36,7 +38,7 @@ var migrationTable = "status_go_" + sqlcipher.DefaultMigrationsTable
 // untilVersion, for testing purposes optional parameter, can be used to limit the migration to a specific version.
 // Pass nil to migrate to the latest available version.
-func Migrate(db *sql.DB, resources *bindata.AssetSource, customSteps []PostStep, untilVersion *uint) error {
+func Migrate(db *sql.DB, resources *bindata.AssetSource, customSteps []*PostStep, untilVersion *uint) error {
```
Not sure what the intention was here, but if it was about optimizing, I doubt using reference in the slice will lead to any optimizations.

IvanBelyakoff replied:
Stefan, I had a function `getCustomStepByName(name)` and added parameters field for custom steps. So I passed the parameter to the step in runtime. For it to work I needed a pointer. I dropped some changes, because unused now as I switched to using sql migration files instead,  but left some changes like this one, which by itself does not look as imrovement without the context

stefandunca commented on walletdatabase/migrations/sql/1691753758_initial.up.sql:
```diff
@@ -0,0 +1,191 @@
+CREATE TABLE blocks (
```
Excellent reference and summarization of the wallet DB structure requirements.

stefandunca reviewed (approved):
Great job. I am looking forward to seeing the results in practice.

IvanBelyakoff committed:
Fixed keystore reencryption failure on password change.
Added a test case for keystore decryption after password change.
Revert the app DB password on wallet password change failure

IvanBelyakoff committed:
Update generated go migration files
